src/main/java/org/icgc_argo/workflow_graph_lib/workflow/client/RdpcClient.java

```diff
@@ -207,9 +208,29 @@ private static RequestEngineParameters engineParamsAdapter(
       return Optional.empty();
     }
 
-    val donorIds =
+    val analysisSamples =
         analysis.getDonors().orElseGet(Collections::emptyList).stream()
-            .map(d -> d.getDonorId().orElse(""))
+            .flatMap(
+                d ->
+                    d.getSpecimens().orElseGet(Collections::emptyList).stream()
+                        .flatMap(
+                            s ->
+                                s.getSamples().orElseGet(Collections::emptyList).stream()
+                                    .map(
+                                        sam ->
+                                            AnalysisSample.newBuilder()
+                                                .setSampleId(sam.getSampleId().orElse(""))
+                                                .setSubmitterSampleId(
+                                                    sam.getSubmitterSampleId().orElse(""))
+                                                .setSubmitterDonorId(
+                                                    d.getSubmitterDonorId().orElse(""))
+                                                .setDonorId(d.getDonorId().orElse(""))
+                                                .setSubmitterSpecimenId(
+                                                    s.getSubmitterSpecimenId().orElse(""))
+                                                .setSpecimenId(s.getSpecimenId().orElse(""))
+                                                .setTumourNormalDesignation(
+                                                    s.getTumourNormalDesignation().orElse(""))
+                                                .build())))
```

Comment:
d == donors? s == specimens? sam == samples? ... we know this but future us and new people will not

Reply:
thats fair. I'll rename them to donor specimen and sample